spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/hypermedia/LinksSnippet.java

```diff
@@ -50,8 +50,8 @@ public LinksSnippet failOnUndocumentedFields(boolean failOnUndocumentedFields) {
     }
 
     @Override
-    protected Type getType(HandlerMethod method) {
-        return documentationType;
+    protected Type[] getType(HandlerMethod method) {
+        return documentationType == null ? null : new Type[] {documentationType};
```

Comment:
An alternative here could be `Optional.ofNullable(documentationType).map(type -> new Type[]{type}).orElse(null)`, but I wasn't sure if you like that fancy way.